src/libsync/logger.cpp

```diff
@@ -148,7 +149,15 @@ void Logger::setLogFile(const QString &name)
     }
 
     bool openSucceeded = false;
+    auto codec = "UTF-8";
     if (name == QLatin1String("-")) {
+#ifdef Q_OS_WIN
+        if (AttachConsole(ATTACH_PARENT_PROCESS)) {
+            freopen("CONOUT$", "w", stdout);
+        }
+        _setmode(_fileno(stdout), _O_U8TEXT);
+        codec = "UTF-16";
```

Comment:
Needed to prevent a crash in ucrt `buf.size() % 2 == 0`

Reply:
This snippet doesn't make a lot of sense to me. Why first enable UTF-8 (I suppose that's the case), then use UTF-16? If this is some "Windows thing", please put a comment above this to explain what you're doing here. A link to wherever you found this snippet would also be nice to have.

Also, you must check the return value of `_setmode`. It might return a non-zero value.

Reply:
Looks like that wasn't needed, only my initial piping to Write-Host broke the encoding.